vision/detect_faces.rb

```diff
@@ -12,61 +12,77 @@
 
-  image.faces.each do |face|
-    puts "Joy:      #{face.likelihood.joy?}"
-    puts "Anger:    #{face.likelihood.anger?}"
-    puts "Sorrow:   #{face.likelihood.sorrow?}"
-    puts "Surprise: #{face.likelihood.surprise?}"
+  content = File.binread image_path
+  image = { content: content }
+  feature = { type: :FACE_DETECTION }
+  request = { image: image, features: [feature] }
+
+  response = image_annotator.batch_annotate_images([request])
+  response.responses.each do |res|
+    res.face_annotations.each do |face|
+      puts "Joy:      #{face.joy_likelihood}"
+      puts "Anger:    #{face.anger_likelihood}"
+      puts "Sorrow:   #{face.sorrow_likelihood}"
+      puts "Surprise: #{face.surprise_likelihood}"
+    end
   end
   # [END vision_face_detection]
 end
 
 # This method is a duplicate of the above method, but with a different
 # description of the 'image_path' variable, demonstrating the gs://bucket/file
 # GCS storage URI format.
-def detect_faces_gcs project_id:, image_path:
+def detect_faces_gcs image_path:
   # [START vision_face_detection_gcs]
-  # project_id = "Your Google Cloud project ID"
   # image_path = "Google Cloud Storage URI, eg. 'gs://my-bucket/image.png'"
 
   require "google/cloud/vision"
 
-  vision = Google::Cloud::Vision.new project: project_id
-  image  = vision.image image_path
+  image_annotator = Google::Cloud::Vision::ImageAnnotator.new
+
+  source = { gcs_image_uri: image_path }
+  image = { source: source }
+  feature = { type: :FACE_DETECTION }
+  request = { image: image, features: [feature] }
 
-  image.faces.each do |face|
-    puts "Joy:      #{face.likelihood.joy?}"
-    puts "Anger:    #{face.likelihood.anger?}"
-    puts "Sorrow:   #{face.likelihood.sorrow?}"
-    puts "Surprise: #{face.likelihood.surprise?}"
+  response = image_annotator.batch_annotate_images([request])
+  response.responses.each do |res|
+    res.face_annotations.each do |face|
+      puts "Joy:      #{face.joy_likelihood}"
+      puts "Anger:    #{face.anger_likelihood}"
+      puts "Sorrow:   #{face.sorrow_likelihood}"
+      puts "Surprise: #{face.surprise_likelihood}"
+    end
   end
   # [END vision_face_detection_gcs]
 end
 
 if __FILE__ == $PROGRAM_NAME
+  require "uri"
+
   image_path = ARGV.shift
-  project_id = ENV["GOOGLE_CLOUD_PROJECT"]
 
-  if image_path
-    detect_faces image_path: image_path, project_id: project_id
-  else
-    puts <<-usage
-Usage: ruby detect_faces.rb [image file path]
+  unless image_path
+    return puts <<-USAGE
+    Usage: ruby detect_faces.rb [image file path]
 
-Example:
-  ruby detect_faces.rb image.png
-  ruby detect_faces.rb https://public-url/image.png
-  ruby detect_faces.rb gs://my-bucket/image.png
-    usage
+    Example:
+      ruby detect_faces.rb image.png
+      ruby detect_faces.rb https://public-url/image.png
+      ruby detect_faces.rb gs://my-bucket/image.png
+    USAGE
   end
+  if image_path =~ URI::DEFAULT_PARSER.new.make_regexp
```

Comment:
Does this work for GCS URI paths as well?

Reply:
yes

Reply:
Shouldn't this be `URI::DEFAULT_PARSER.make_regexp`? (Or, more simply, `URI.regexp`?)

Calling `URI::DEFAULT_PARSER.new.make_regexp` throws `NoMethodError (undefined method 'new' for #<URI::RFC2396_Parser:0x0000123456798>)` for me.